trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -112,41 +114,356 @@ def spin(self,conditions):
+                       (pi*U*(Ua*cos_psi - Ut*sin_psi)*(beta - np.arctan((Wa+Wa)/(Wt+Wt))))/(2.*(f_wt_2 + f_wa_2)**(0.5))
+                       + (pi*U*(f_wt_2 +f_wa_2)**(0.5)*(U + Utcospsi  +  Uasinpsi))/(2.*(f_wa_2/(f_wt_2) + 1.)*utpUcospsi2)
+                       - (4.*U*piece*((16.*UapUsinpsi2)/(BB*pi2*f_wt_2) + 1.)**(0.5)*(R - r)*(Ut/2. - 
+                      (Ucospsi)/2.)*(U + Utcospsi + Uasinpsi ))/(f_wa_2*(1. - np.exp(-(B*(Wt+Wt)*(R - 
+                       r))/(r*(Wa+Wa))))**(0.5)) + (128.*U*r*arccos_piece*(Wa+Wa)*(Ut/2. - (Ucospsi)/2.)*(U + 
+                       Utcospsi  + Uasinpsi ))/(BBB*pi2*utpUcospsi*utpUcospsi2*((16.*f_wa_2)/(BB*pi2*f_wt_2) + 1.)**(0.5))) 
+            
+            dR_dpsi[np.isnan(dR_dpsi)] = 0.1
+                      
+            dpsi   = -Rsquiggly/dR_dpsi
+            psi    = psi + dpsi
+            diff   = np.max(abs(psiold-psi))
+            psiold = psi
+            
+            # If its really not going to converge
+            if np.any(psi>(pi*85.0/180.)) and np.any(dpsi>0.0):
+                break
+                
+            ii+=1
+                
+            if ii>2000:
+                broke = True
+                break
+        
+        #There is also RE scaling
+        #This is an atrocious fit of DAE51 data at RE=50k for Cd
+        Cdval = (0.108*(Cl*Cl*Cl*Cl)-0.2612*(Cl*Cl*Cl)+0.181*(Cl*Cl)-0.0139*Cl+0.0278)*((50000./Re)**0.2)
+        Cdval[alpha>=pi/2] = 2.
+        
+        #More Cd scaling from Mach from AA241ab notes for turbulent skin friction
+        Tw_Tinf = 1. + 1.78*(Ma*Ma)
+        Tp_Tinf = 1. + 0.035*(Ma*Ma) + 0.45*(Tw_Tinf-1.)
+        Tp      = (Tp_Tinf)*T
+        Rp_Rinf = (Tp_Tinf**2.5)*(Tp+110.4)/(T+110.4)
+        
+        Cd = ((1/Tp_Tinf)*(1/Rp_Rinf)**0.2)*Cdval 
+        
+        epsilon  = Cd/Cl
+        epsilon[epsilon==np.inf] = 10. 
+        deltar   = (r[1]-r[0])
+        thrust   = rho*B*(np.sum(Gamma*(Wt-epsilon*Wa)*deltar,axis=1)[:,None])
+        torque   = rho*B*np.sum(Gamma*(Wa+epsilon*Wt)*r*deltar,axis=1)[:,None]
+        D        = 2*R 
+        Ct       = thrust/(rho*(n*n)*(D*D*D*D))
+        Ct[Ct<0] = 0.        #prevent things from breaking
+        kappa    = self.induced_power_factor 
+        Cd0      = self.profile_drag_coefficient   
+        Cp    = np.zeros_like(Ct)
+        power = np.zeros_like(Ct)        
+        for i in range(len(Vv)):
+            if -1. <Vv[i][0] <1.: # vertical/axial flight
+                Cp[i]       = (kappa*(Ct[i]**1.5)/(2**.5))+sigma*Cd0/8.
+                power[i]    = Cp[i]*(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D))
+                torque[i]   = power[i]/omega[i]  
+            else:  
+                power[i]    = torque[i]*omega[i]   
+                Cp[i]       = power[i]/(rho[i]*(n[i]*n[i]*n[i])*(D*D*D*D*D))
+
+
+        
+  
+        thrust[conditions.propulsion.throttle[:,0] <=0.0] = 0.0
```

Comment:
Could this cause segment convergence issues?

Reply:
I am not sure. Once again this syntax was not changed from the previous script. It only appears like new code since @planes  added `spin_surrogate` and `spin_variable_pitch` functions . I plan to look at convergence issues in another PR